src/core/core.scale.js

```diff
@@ -726,18 +726,19 @@ module.exports = Element.extend({
 	/**
 	 * Actually draw the scale on the canvas
 	 * @param {object} chartArea - the area of the chart to draw full grid lines on
+	 * @param {boolean} top - drawing on top / after datasets
 	 */
-	draw: function(chartArea) {
+	draw: function(chartArea, top) {
 		var me = this;
 		var options = me.options;
+		var optionTicks = options.ticks;
 
-		if (!me._isVisible()) {
+		if (!me._isVisible() || (top && optionTicks.display !== 'top')) {
```

Comment:
I'd probably have done this check in `core.controller` instead

Reply:
Not sure its worth digging in to `box` to see if its a scale and if its ticks should be drawn above in `core.controller`.
OTOH custom scales that don't implement this would be drawn twice now. 
I'll look into this.

`above` instead of `top` makes more sense to me too.